ci/test_wheel_cugraph.sh

```diff
@@ -11,12 +11,4 @@ python -m pip install --no-deps ./local-pylibcugraph-dep/pylibcugraph*.whl
 # Always install latest dask for testing
 python -m pip install git+https://github.com/dask/dask.git@main git+https://github.com/dask/distributed.git@main
 
-# Only download test data for x86
-arch=$(uname -m)
-if [[ "${arch}" == "x86_64" ]]; then
-    pushd ./datasets
-    bash ./get_test_data.sh
-    popd
-fi
```

Comment:
I guess this change is necessary for nightlies? Were nightly tests for arm not running the tests that required datasets?

Reply:
We shouldn't remove this, but might need to modify it to only be skipped in ARM PR jobs. In ARM PR jobs, we can only run "smoke tests" due to limited resource capacity.

https://github.com/rapidsai/cugraph/blob/5c34d3dd340c76678d8f2667057c6b0ce2f1f480/ci/test_wheel.sh#L16

Downloading the datasets is time-intensive and should not be performed for "smoke tests" in ARM PR jobs because it wastes a lot of time on the ARM GPU node.

Reply:
I'm terribly sorry. I read this backwards. This removed downloading datasets for ALL jobs (all arches, nightly/PR). Are we not dependent on datasets for tests to pass anymore?

Reply:
The `get_test_data.sh` script will download supplemental datasets that are too big to commit to our github repo.  Those are currently only used by C++ tests.  The python tests use the smaller .csv datasets committed to the repo [here](https://github.com/rapidsai/cugraph/tree/branch-23.10/datasets), so it's safe to skip downloading when only testing python code.

I only removed it here since it was in the proximity of the changes for the wheel builds, but I think an audit of what runs require downloading the supplemental datasets would be good for a separate PR.

Prior to the recent commits to the PR which triggered CI running now, we had everything passing with the download step removed.

Reply:
Thanks @rlratzel, I'll approve now.